### PR TITLE
feat: added support for v3 config file

### PR DESCRIPTION
**Pull request by isaackps**
# In this PR:
- Added support for v3 config file. 

# Stories
https://www.pivotaltracker.com/n/projects/2424361/stories/180703033

### CODE REVIEW

File: src/implementations/config/create.ts [hunk 1]
```diff
@@ -35,9 +37,17 @@ export const create = async ({
...
   const hasTransferableRecord = forms.some((form) => form.type === "TRANSFERABLE_RECORD");
   const hasDocumentStore = forms.some((form) => form.type === "VERIFIABLE_DOCUMENT");
-  const hasDid = forms.some(
-    (form) => form.defaults.issuers.some((issuer) => issuer.identityProof?.type.includes("DID")) // TODO: v3 does not have `issuers` but `issuer` instead
-  );
+  const hasDid = forms.some((form) => {
+    //check form for v2/v3
+    if (isV3Form(form.defaults)) {
+      const v3Defaults = form.defaults as v3.OpenAttestationDocument;
+      const didCheckList = ["DID", "DNS-DID"];
```
Comment (simboonlong) [inline src/implementations/config/create.ts:44]: `utils.isRawV3Document(form.defaults)` will that work? -> https://github.com/Open-Attestation/open-attestation/blob/master/src/shared/utils/guard.ts#L30

File: src/implementations/config/create.ts [hunk 2]
```diff
@@ -35,9 +37,17 @@ export const create = async ({
...
+      return didCheckList.includes(v3Defaults.openAttestationMetadata.proof.method);
+    } else {
+      const v2Defaults = form.defaults as v2.OpenAttestationDocument;
+      return v2Defaults.issuers.some((issuer) => issuer.identityProof?.type.includes("DID"));
+    }
```
Comment (simboonlong) [inline src/implementations/config/create.ts:49]: how about `DNS-DID`

File: src/implementations/config/helpers.ts [hunk 3]
```diff
@@ -28,6 +28,10 @@ interface UpdatedForms {
   dnsTransferableRecord: Dns;
 }
 
+export const isV3Form = (formDefaults: OpenAttestationDocument): boolean => {
+  return "@context" in formDefaults && "openAttestationMetadata" in formDefaults;
+};
+
 export const getConfigWithUpdatedForms = ({
   configFile,
```
Comment (simboonlong) [inline src/implementations/config/helpers.ts:36]: this is almost the same as https://github.com/TradeTrust/tradetrust-config/blob/master/src/helpers/helpers.ts#L20 and https://github.com/TradeTrust/tradetrust-config/blob/master/src/helpers/helpers.ts#L83

should we move those helpers from tradetrust-config to here instead, then export out so we can share the same. wdyt

File: src/implementations/config/create.ts (hunk 1 above)
Comment (isaackps) [reply src/implementations/config/create.ts:44]: haha ok will update, forgot about that.

File: src/implementations/config/create.ts (hunk 2 above)
Comment (isaackps) [reply src/implementations/config/create.ts:49]: ok will update it

File: src/implementations/config/helpers.ts (hunk 3 above)
Comment (isaackps) [reply src/implementations/config/helpers.ts]: ok

**Review by simboonlong — APPROVED** [review]
lgtm

**Pull request merged**